x-pack/plugins/ingest_manager/server/services/agent_policy_update.ts

```diff
@@ -17,20 +33,25 @@ export async function agentPolicyUpdateEventHandler(
 ) {
   const adminUser = await outputService.getAdminUser(soClient);
   const outputId = await outputService.getDefaultOutputId(soClient);
+
   // If no admin user and no default output fleet is not enabled just skip this hook
   if (!adminUser || !outputId) {
     return;
   }
 
+  // `soClient` from ingest `appContextService` is used to create policy change actions
+  // to ensure encrypted SOs are handled correctly
+  const internalSoClient = appContextService.getInternalUserSOClient(fakeRequest);
```

Comment:
the alternative to this would be to include the encrypted SO plugin in Security_Solution?

@neptunian FYI, is this what would be needed by the error you were seeing?  I think it had to do with api key generation?

Reply:
@kevinlog yes, but I don't know much about how that could/would impact the creation of the artifacts as well as the Exceptions List since the current soClient seems to be passed on to those other areas as well as Manifest manger. Really need @madirey (whenever she has some time) to comment on this. 

Another alternative might be to use a encryptedSavedObject client only for manifest manager, since I think that is the only one that uses the Ingest policy services. I looked at how Ingest creates their soClient and I think we have what is needed in security solution `plugin.start()` to create it.